catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ValidateCommand.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.commands.catalog;
+
+import static org.apache.karaf.shell.support.ansi.SimpleAnsi.COLOR_CYAN;
+import static org.apache.karaf.shell.support.ansi.SimpleAnsi.COLOR_DEFAULT;
+import static org.apache.karaf.shell.support.ansi.SimpleAnsi.COLOR_RED;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Argument;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Completion;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.support.completers.FileCompleter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.util.Describable;
+import ddf.catalog.validation.MetacardValidator;
+import ddf.catalog.validation.ValidationException;
+
+@Command(scope = CatalogCommands.NAMESPACE, name = "validate", description = "Validates an XML file against all installed validators.")
+@Service
+public class ValidateCommand implements Action {
+
+    @Argument(index = 0, name = "fileName", description = "The path to the file that you want to validate", required = true, multiValued = false)
+    @Completion(FileCompleter.class)
+    String filename;
+
+    @Reference
+    List<MetacardValidator> validators;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ValidateCommand.class);
+
+    protected PrintStream console = System.out;
+
+    private static final String INDENT =       "|   ";
+    private static final String LAST_INDENT =  "    ";
+    private static final String ELEMENT =      "+-- ";
+    private static final String LAST_ELEMENT = "`-- ";
+
+    private static final String WARNING_TITLE = "Warnings";
+    private static final String ERROR_TITLE =   "Errors";
+
+    private static final String WARNING_COLOR = COLOR_CYAN;
+    private static final String ERROR_COLOR =   COLOR_RED;
+
+    public Object execute() throws Exception {
```

Comment:
I was wondering why does `execute()` always return null? When not just make it a void-return function?